sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -2493,18 +2480,30 @@ bool doesDevSupportDeviceRequirements(const device &Dev,
         DeviceBinaryProperty(*(ReqdWGSizePropIt.value())).asByteArray();
     // Drop 8 bytes describing the size of the byte array.
     ReqdWGSize.dropBytes(8);
-    int ReqdWGSizeAllDimsTotal = 1;
-    std::vector<int> ReqdWGSizeVec;
+    uint64_t ReqdWGSizeAllDimsTotal = 1;
+    std::vector<uint64_t> ReqdWGSizeVec;
     int Dims = 0;
     while (!ReqdWGSize.empty()) {
-      int SingleDimSize = ReqdWGSize.consume<int>();
-      ReqdWGSizeAllDimsTotal *= SingleDimSize;
+      // The reqd_work_group_size data is stored as uint32_t's,
+      // but we'll widen the result to uint64_t.
+      uint64_t SingleDimSize = ReqdWGSize.consume<uint32_t>();
```

Comment:
If the spec says that `reqd-work-group-size` should support `size_t` values (KhronosGroup/SYCL-Docs#450), then we shouldn't store the attribute values in any 32-bit type at all.

Note that changing value type of the property would be an ABI-break, because we won't be able to read properties from binaries compiled using older compilers.

To workaround this, I would suggest to introduce a new `reqd_work_group_size_size_t` (or something like that) property. Compiler will always generate it, but runtime will check for both and use whichever is present, doing widening for older property. Once the next ABI-breaking window is open, we will be able to drop `_size_t` property and simply change values type of the original one.